Core.ApplicationServices/Organizations/Write/OrganizationWriteService.cs

```diff
@@ -45,7 +45,7 @@ public Result<Organization, OperationError> UpdateMasterData(Guid organizationUu
     {
         using var transaction = _transactionManager.Begin();
         var result = _organizationService.GetOrganization(organizationUuid)
-            .Bind(organization => Update(organization, parameters));
+            .Bind(organization => UpdateMasterDataIfWriteAccess(organization, parameters));
```

Comment:
Instead of the UpdateIfWriteAccess methods I would change them to GetOrganizationAndVerifyWriteAccess method, then you can just Bind to that and do the PerformUpdates, or then verify the cvr and perform org update.

Also, it's a small thing, and I can't really justify it, but the conditional method name feels off to me. Maybe, because it implies 2 different outcomes